common/src/main/java/com/microsoft/identity/common/internal/net/ObjectMapper.java

```diff
@@ -100,6 +101,16 @@ public static String serializeObjectToFormUrlEncoded(Object object) throws Unsup
         Type stringMap = new TypeToken<TreeMap<String, String>>() {
         }.getType();
         TreeMap<String, String> fields = new Gson().fromJson(json, stringMap);
+        if (object instanceof IHasExtraParameters) {
+            IHasExtraParameters params = (IHasExtraParameters) object;
+            if (params.getExtraParameters() != null) {
+                for (Map.Entry<String, String> e : params.getExtraParameters()) {
+                    if (e.getKey() != null) {
+                        fields.put(e.getKey(), e.getValue());
+                    }
+                }
+            }
+        }
 
         StringBuilder builder = new StringBuilder();
```

Comment:
```suggestion
        final StringBuilder builder = new StringBuilder();
```

(Not part of your PR, but why not 😏)